File: prac4.cpp
#include <bits/stdc++.h>
using namespace std;

int max(int a, int b)
{
    return (a > b) ? a : b;
}

int kSpack(int W, int wt[], int val[], int n)
{
    if (n == 0 || W == 0)
        return 0;
    // If weight of the item is more than capacity then this item cannot be included in knapsack
    if (wt[n - 1] > W)
        return kSpack(W, wt, val, n - 1);
    else
        return max(val[n - 1] + kSpack(W - wt[n - 1], wt, val, n - 1), kSpack(W, wt, val, n - 1));
}
int main()
{
    int profit[] = {60, 100, 120};
    int weight[] = {10, 20, 30};
    int W = 50;
    int n = sizeof(profit) / sizeof(profit[0]);
    cout << "Maximum value that can be put in a knapsack of capacity W is: " << kSpack(W, weight, profit, n);
    return 0;
}

File: prac4.py
def KsK(capacity,profit,wt,n):
    k = [[ 0 for i in range(capacity + 1)] for x in range( n + 1)]
    print(capacity)
    for i in range(n + 1):
        for w in range(capacity + 1):
            if i == 0 or w == 0:
                k[i][w] = 0
            elif wt[i - 1] <= w:
                k[i][w] = max(profit[i-1]+k[i-1][w-wt[i-1]],k[i-1][w])
            else:
                k[i][w] = k[i - 1][w]
    return k[n][capacity]

capacity = 50
profit = [60,100,120]
wt = [10,20,30]
n=len(profit)
print(KsK(capacity, profit, wt, n))
